Assets/MixedRealityToolkit/Utilities/MixedRealityServiceRegistry.cs

```diff
@@ -0,0 +1,222 @@
+        /// </returns>
+        public static bool RemoveService<T>(T serviceInstance, IMixedRealityServiceRegistrar registrar) where T : IMixedRealityService
+        {
+            return RemoveServiceInternal(typeof(T), serviceInstance, registrar);
+        }
+
+        /// <summary>
+        /// Removes an <see cref="IMixedRealityService"/> instance from the registry.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service being removed.</typeparam>
+        /// <param name="serviceInstance">Instance of the service to remove.</param>
+        /// <returns>
+        /// True if the service was successfully removed, false otherwise.
+        /// </returns>
+        public static bool RemoveService<T>(T serviceInstance) where T : IMixedRealityService
+        {
+            T tempService;
+            IMixedRealityServiceRegistrar registrar;
+
+            if (!TryGetService<T>(out tempService, out registrar))
+            {
+                return false;
+            }
+
+            if (!object.ReferenceEquals(serviceInstance, tempService))
+            {
+                return false;
+            }
+
+            return RemoveServiceInternal(typeof(T), serviceInstance, registrar);
+        }
+
+        /// <summary>
+        /// Removes an <see cref="IMixedRealityService"/> instance from the registry.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service being removed.</typeparam>
+        /// <param name="name">The friendly name of the service to remove.</param>
+        /// <returns>
+        /// True if the service was successfully removed, false otherwise.
+        /// </returns>
+        public static bool RemoveService<T>(string name) where T : IMixedRealityService
+        {
+            T tempService;            
+            IMixedRealityServiceRegistrar registrar;
+
+            if (!TryGetService<T>(out tempService, out registrar, name))
+            {
+                return false;
+            }
+
+            return RemoveServiceInternal(typeof(T), tempService, registrar);
+        }
+
+        /// <summary>
+        /// Removes an <see cref="IMixedRealityService"/> instance from the registry.
+        /// </summary>
+        /// <param name="interfaceType">The interface type of the service being removed.</param>
+        /// <param name="serviceInstance">Instance of the service to remove.</param>
+        /// <param name="registrar">Instance of the registrar manages the service.</param>
+        /// <returns>
+        /// True if the service was successfully removed, false otherwise.
+        /// </returns>
+        private static bool RemoveServiceInternal(
+            Type interfaceType,
+            IMixedRealityService serviceInstance,
+            IMixedRealityServiceRegistrar registrar)
+        {
+            List<KeyValuePair<IMixedRealityService, IMixedRealityServiceRegistrar>> services = registry[interfaceType];
```

Comment:
I think this will throw a KeyNotFound exception if this doesn't exist. AFAICT there are public->private paths that don't guard against this.

Reply:
you're right! i'll check for that

Reply:
fixed